fretboard/src/main/java/mozilla/components/service/fretboard/ExperimentSource.kt

```diff
@@ -0,0 +1,22 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.fretboard
+
+/**
+ * Represents a location where experiments are stored
+ * (Kinto, a JSON file on a server, etc)
+ */
+interface ExperimentSource {
+    /**
+     * Requests new experiments from the source,
+     * parsing the response into experiments
+     *
+     * @param client Http client to use, provided by Fretboard
+     * @param experiments list of already downloaded experiments
+     * (in order to process a diff response, for example)
+     * @return modified list of experiments
+     */
+    fun getExperiments(client: HttpClient, experiments: List<Experiment>): List<Experiment>
```

Comment:
This signature may be problematic if you pull diffs from Kinto. Is this something you want to add at a later time?

For a deleted item Kinto may only return this:
```
        {
            "deleted": true,
            "id": "a5f490b2-218e-4d71-ac5a-f046ae285c55",
            "last_modified": 1434644823180
        }
```

Reply:
Yes, actually pulling diffs from Kinto is already implemented inside the KintoExperimentSource class. The experiments parameter of the getExperiments function are the experiments from local storage (the already downloaded ones). In the case of a deleted response, as shown above, KintoExperimentSource removes this element from the list of elements that came from the storage and returns the remaining list

Reply:
Okay, great! :)